### PR TITLE
feat(headless/commerce): add basic plp, search and recs use case examples

### PR DESCRIPTION
**Pull request by Spuffynism**
I also exposed the sample commerce engine configuration in headless.

[CAPI-918]

[CAPI-918]: https://coveord.atlassian.net/browse/CAPI-918?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/headless/src/app/commerce-engine/commerce-engine-configuration.ts [hunk 1]
```diff
@@ -40,24 +41,49 @@ export const commerceEngineConfigurationSchema =
 
 export function getSampleCommerceEngineConfiguration(): CommerceEngineConfiguration {
```
Comment (Spuffynism) [inline packages/headless/src/app/commerce-engine/commerce-engine-configuration.ts:42]: I copied this configuration over from atomic. This access token was already exposed in our documentation. https://github.com/coveo/ui-kit/blob/7aa6094d4bf63a81ae88c7327f84cc808f92f4b6/packages/atomic/storybookUtils/commerce-interface-wrapper.tsx#L10-L17

Comment (louis-bompart) [reply packages/headless/src/app/commerce-engine/commerce-engine-configuration.ts:42]: I don't really fancy exposing a dev org....
At the very least, let's put a TODO with KIT-3244

Comment (Spuffynism) [reply packages/headless/src/app/commerce-engine/commerce-engine-configuration.ts:42]: Would you rather we expose the prod barca org? It also has a public access token.

Nonetheless, I'll add a TODO with KIT-3244.

**Review by olamothe — APPROVED** [review]

File: packages/headless/src/app/commerce-engine/commerce-engine-configuration.ts (hunk 1 above)
Comment (Spuffynism) [reply packages/headless/src/app/commerce-engine/commerce-engine-configuration.ts:43]: We decided to keep the dev org for now to not potentially mess up barca prod analytics.

File: packages/headless/src/app/commerce-engine/commerce-engine-configuration.ts [hunk 2]
```diff
@@ -38,26 +39,52 @@ export const commerceEngineConfigurationSchema =
     }),
   });
 
+// TODO KIT-3244: Use a different sample organization
 export function getSampleCommerceEngineConfiguration(): CommerceEngineConfiguration {
   return {
-    organizationId: 'fashioncoveodemocomgzh7iep8',
-    // deepcode ignore HardcodedNonCryptoSecret: Public key freely available for our documentation
-    accessToken: 'xx149e3ec9-786f-4c6c-b64f-49a403b930de',
+    accessToken: 'xxc481d5de-16cb-4290-bd78-45345319d94c',
```
Comment (pvanool) [inline packages/headless/src/app/commerce-engine/commerce-engine-configuration.ts:45]: I'm a bit surprised that we commit `accesToken`s. Probably expired anyway?

**Review by pvanool — APPROVED** [review]
LGTM

File: packages/headless/src/app/commerce-engine/commerce-engine-configuration.ts (hunk 2 above)
Comment (louis-bompart) [reply packages/headless/src/app/commerce-engine/commerce-engine-configuration.ts:45]: No they aint: But some access token that are solely made for search are OK.

File: packages/samples/headless-react/src/pages/CommercePage.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  const search = buildSearch(engine);
+  const recommendations = buildRecommendations(engine, {
+    options: {
+      slotId: 'abccdea4-7d8d-4d56-b593-20267083f88f',
+    },
+  });
+
+  return (
+    <AppContext.Provider value={{commerceEngine: engine}}>
+      <Section title="product-listing">
```
Comment (fbeaudoincoveo) [inline packages/samples/headless-react/src/pages/CommercePage.tsx:34]: It doesn't make sense to have both product listing and search in the same page; I don't think we should showcase that in our examples.

**Review by fbeaudoincoveo — APPROVED** [review]

**Review by BenKandelaars — APPROVED** [review]
Nice

**Comment by Spuffynism** [comment]
@fbeaudoincoveo In https://github.com/coveo/ui-kit/pull/4079/commits/0020536b8d270e6776f7e63d657b958a7fc606d3, I put the commerce solution type under a `CommerceApp`, which holds solution-type specific pages. This represents more closely how controllers should actually be used.

https://github.com/coveo/ui-kit/assets/8978908/0d358237-4d04-4c55-b173-d80f33b9e68e

**Review by fbeaudoincoveo — APPROVED** [review]

**Pull request merged**